upi/vsphere/ipam/variables.tf

```diff
@@ -0,0 +1,33 @@
+variable "name" {
```

Comment:
I think the inputs should be

```
ipam
ipam_api_token

machine_cidr // for getting the network
hostname list(string) // all the hostname for which we need an ip.
ip_addresses list(string) // so that you can return them as-is and not call the ipam when >0
```